src/main/java/edu/iu/terracotta/service/app/impl/OutcomeServiceImpl.java

```diff
@@ -189,12 +191,24 @@ public boolean outcomeBelongsToExperimentAndExposure(Long experimentId, Long exp
     @Override
     public List<OutcomePotentialDto> potentialOutcomes(Long experimentId) throws DataServiceException, CanvasApiException {
         Optional<Experiment> experiment = experimentService.findById(experimentId);
+        List<Assignment> assignmentList = allRepositories.assignmentRepository.findByExposure_Experiment_ExperimentId(experimentId);
         List<OutcomePotentialDto> outcomePotentialDtos = new ArrayList<>();
         if (experiment.isPresent()) {
             String canvasCourseId = StringUtils.substringBetween(experiment.get().getLtiContextEntity().getContext_memberships_url(), "courses/", "/names");
-            List<AssignmentExtended> assignmentExtendedList = canvasAPIClient.listAssignments(canvasCourseId,experiment.get().getPlatformDeployment());
-            for (AssignmentExtended assignmentExtended:assignmentExtendedList){
+            List<AssignmentExtended> assignmentExtendedList = canvasAPIClient.listAssignments(canvasCourseId, experiment.get().getPlatformDeployment());
+            for (AssignmentExtended assignmentExtended : assignmentExtendedList) {
+                List<Assignment> matched = assignmentList.stream().filter(x -> {
+                    if (assignmentExtended.getId().intValue() == Integer.valueOf(x.getLmsAssignmentId()).intValue()) {
+                        return true;
+                    } else if (experiment.get().getConsentDocument() != null && assignmentExtended.getId().intValue() ==
```

Comment:
The logic here is kind of weird. The consent assignment won't be in the assignmentList so it doesn't make sense to check if assignmentExtended is a consent assignment within a filter of assignmentList.

I think it would be clearer if the consent assignment check were a separate one from the `assignmentList.stream().filter()`